### PR TITLE
Provide cancellation token to SP mappers

### LINKED ISSUE #577: Feature request: provide CancellationToken to SP mappers
ExecuteSqlRawAsync has the overloaded implementation:
```C#
public static async Task<int> ExecuteSqlRawAsync(
    [NotNull] this DatabaseFacade databaseFacade,
    [NotNull] string sql,
    [NotNull] IEnumerable<object> parameters,
    CancellationToken cancellationToken = default)
```
Is that possible to create mapping methods with CancellationToken as well? Method signature might be the following:
```C#
public async Task<int> LONG_RUNNING_SP(int? param1, int? param2, 
    OutputParameter<int> returnValue = null, 
    CancellationToken cancellationToken = default)
```

### PR DESCRIPTION
**Pull request by tiomny**
Resolve #577.

### CODE REVIEW

File: src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ namespace RevEng.Core.Procedures
 {
     public class SqlServerStoredProcedureScaffolder : IProcedureScaffolder
     {
+        private const string parameterPreffix = "parameter";
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs:24]: Preffix => Prefix

File: src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs [hunk 2]
```diff
@@ -231,14 +255,14 @@ private void GenerateProcedure(Procedure procedure, ProcedureModel model)
...
-                .Select(p => $"parameter{p.Name}");
+                .Select(p => $"{parameterPreffix}{p.Name}");
 
             var paramList = procedure.Parameters
                 .Select(p => p.Output ? $"@{p.Name} OUTPUT" : $"@{p.Name}").ToList();
 
             paramList.RemoveAt(paramList.Count - 1);
 
-            var fullExec = $"\"EXEC @{retValueName} = [{procedure.Schema}].[{procedure.Name}] {string.Join(", ", paramList)}\", {string.Join(", ", paramNames)}".Replace(" \"", "\"");
+            var fullExec = $"\"EXEC @{retValueName} = [{procedure.Schema}].[{procedure.Name}] {string.Join(", ", paramList)}\", sqlParameters, cancellationToken".Replace(" \"", "\"");
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs:265]: Where is "sqlParameters"defined?

File: src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs [hunk 3]
```diff
@@ -271,13 +295,22 @@ private static string GenerateMethodSignature(Procedure procedure, List<Procedur
...
         }
 
+        private void GenerateParametersVar(ProcedureParameter parameter)
+        {
+            _sb.Append($"var {parameterPreffix}{parameter.Name} = ");
+            GenerateParameters(parameter);
+            _sb.AppendLine(";");
+        }
+
         private void GenerateParameters(ProcedureParameter parameter)
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs:311]: Rename to GenerateParameter

**Comment by ErikEJ** [comment]
Have you run a manual test?

**Review by ErikEJ — CHANGES_REQUESTED** [review]

File: src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs (hunk 2 above)
Comment (tiomny) [reply src/GUI/RevEng.Core/Procedures/SqlServerStoredProcedureScaffolder.cs:265]: Line 209:
```C#
_sb.AppendLine("var sqlParameters = new []");
```

**Comment by tiomny** [comment]
> 
> 
> Have you run a manual test?

Yes, I have - the generated code was built without errors.

**Review by ErikEJ — APPROVED** [review]

**Comment by ErikEJ** [comment]
Thanks a lot for your contribution

**Pull request merged**